include/openenclave/attestation/sgx/evidence.h

```diff
@@ -152,6 +153,60 @@ OE_EXTERNC_BEGIN
 
 #define OE_CLAIM_SGX_REPORT_DATA "sgx_report_data"
 
+/**
+ * TCB level status of SGX platform. This enumeration type defines return codes
+ * for SGX TCB status, which is the claim value for ::OE_CLAIM_TCB_STATUS.
+ */
+typedef enum _oe_sgx_tcb_status
+{
+    /**
+     * TCB level of SGX platform is up-to-date.
+     */
+    OE_SGX_TCB_STATUS_UP_TO_DATE = 0,
+
+    /**
+     * TCB level of SGX platform is outdated.
+     */
+    OE_SGX_TCB_STATUS_OUT_OF_DATE = 1,
+
+    /**
+     * TCB level of SGX platform is revoked. The platform is not trustworthy.
+     */
+    OE_SGX_TCB_STATUS_REVOKED = 2,
+
+    /**
+     * TCB level of the SGX platform is up-to-date but additional configuration
+     * of SGX platform may be needed.
+     */
+    OE_SGX_TCB_STATUS_CONFIGURATION_NEEDED = 3,
+
+    /**
+     * TCB level of SGX platform is outdated and additional configuration of SGX
+     * platform may be needed.
+     */
+    OE_SGX_TCB_STATUS_OUT_OF_DATE_CONFIGURATION_NEEDED = 4,
+
+    /**
+     * TCB level of the SGX platform is up-to-date but due to certain issues
+     * affecting the platform, additional Software Hardening in the attesting
+     * SGX enclaves may be needed.
+     */
+    OE_SGX_TCB_STATUS_SW_HARDENING_NEEDED = 5,
+
+    /**
+     * TCB level of the SGX platform is up-to-date but additional configuration
+     * for the platform and Software Hardening in the attesting SGX enclaves may
+     * be needed.
+     */
+    OE_SGX_TCB_STATUS_CONFIGURATION_AND_SW_HARDENING_NEEDED = 6,
+
+    /**
+     * TCB level is not valid.
+     */
+    OE_SGX_TCB_STATUS_INVALID = OE_ENUM_MAX,
```

Comment:
Other enums define OE_ENUM_MAX as an internal value that's not meant for external consumption. Here OE_SGX_TCB_STATUS_INVALID is a legit value and should not be defined as OE_ENUM_MAX. I'd recommend just defining it as 7. Future status values will start from 8.